aikau/src/main/resources/alfresco/lists/ItemSelectionMixin.js

```diff
@@ -0,0 +1,252 @@
+/**
+ * Copyright (C) 2005-2015 Alfresco Software Limited.
+ *
+ * This file is part of Alfresco
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * This module should be mixed into renderers that wish to be able to either display selected state or that
+ * can be clicked to change selected state. This is mixed into both the [Selector]{@link module:alfresco/renderers/Selector}
+ * and [Thumbnail]{@link module:alfresco/renderers/Thumbnail} renderers.
+ * 
+ * @module alfresco/lists/ItemSelectionMixin
+ * @author Dave Draper
+ * @since 1.0.40
+ */
+define(["dojo/_base/declare",
+        "alfresco/core/Core",
+        "alfresco/core/topics",
+        "alfresco/documentlibrary/_AlfDocumentListTopicMixin",
+        "dojo/_base/lang",
+        "dojo/_base/array",
+        "dojo/dom-class",
+        "dojo/_base/event"], 
+        function( declare, Core, topics, _AlfDocumentListTopicMixin, lang, array, domClass, Event) {
+   
+   return declare([Core, _AlfDocumentListTopicMixin], {
+      
+      /**
+       * The dot-notation property in the currentItem that uniquely idenfities that item. This defaults
+       * to "nodeRef" as the most likely use case is for working with Nodes but this can be configured
+       * to a different value.
+       *
+       * @instance
+       * @type {string}
+       * @default
+       */
+      itemKey: "nodeRef",
```

Comment:
Is there any mileage in having this as `@type {string|string[]}` so that people could also specify a preference order of properties, e.g. `["nodeRef", "node.nodeRef"]`?